liquibase/liquibase
# fix support for custom DB2Database implementations

Fixes #2058

- - -
## Dev Handoff Notes (Internal Use)
#### Links
* Fixed Issues: 
  * [https://github.com/liquibase/liquibase/issues/2058](https://github.com/liquibase/liquibase/issues/2058)
  * [https://github.com/liquibase/liquibase/issues/2067](https://github.com/liquibase/liquibase/issues/2067)
* Test Results: [https://github.com/liquibase/liquibase/pull/](https://github.com/liquibase/liquibase/pull/){PR ID}/checks

#### Testing
* Steps to Reproduce:
  * Requires a db2i datatabase and the corresponding extension
  * [https://github.com/liquibase/liquibase/issues/2058](https://github.com/liquibase/liquibase/issues/2058)
  * [https://github.com/liquibase/liquibase/issues/2067](https://github.com/liquibase/liquibase/issues/2067)
* Guidance:
  * Impact: fixes a how we are checking the type of database so we correctly use the db2i extension, does not impact the other database support

#### Dev Verification
Reviewed code only, since I don't have a db2i database to test with. But, I see how the fix would address the problem with the extension.

#### Github Users
* wardvijf

## Test Requirements (Liquibase Internal QA)
**Manual Tests**

* Verify the functional tests are passing in the GitHub actions build.
* None required.
* This fix is targeted to a database platform that QA does not have access to at this time.

**Automated Tests**

* No new functional tests required.
* No new test-harness tests required.



┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-2168) by [Unito](https://www.unito.io)
